src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+               cbsd_purge_list = FSSPurgeListRefModel(fad_records,fssOobe)
+ 
+               # Remove purged cbsd from fad record
+               fad_records_iap = removePurgeCbsd(fad_records,cbsd_purge_list)
+            else:
+               fad_records_iap = fad_records
+
+         #Call IAP reference model, fad_records contains fad_UUT and fad_SAS_TH
+         iap_refmodel_margin_list = IapReferenceModel(config['headroom'], pe,fad_records_iap)
+ 
+
+ 
+ 
+       # Step 18,19,20 and 21 : 
+       # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
+       dp.HeartBeatRequestForAllGrants_relinquish()
+
+       #Step 22: Calculating the Aggregate interface and IAP reference model invoke.
+       cbsd_agg_list = dp.GetAllCbsdObjectsWithAtLeastOneGrant()
+       for pe in protectedEntities:
+             
+          #invoking Aggregate Interference Model API,
+          agg_interface_list = AggregateInterferenceModel(cbsd_agg_list,pe)
+
+          #invoking Aggregate Interference API,with IAP reference model EIRP values
+          # This is dummy API, need clarity on the parameters to be passed  
+          agg_interference_iap_list = AggregateInterferenceModelWithIAP(cbsd_agg_list,pe)
+ 
+          #check MCP.1
+          checkMCP_IAP(pe,agg_interface_list,agg_interference_iap_list,pre-IAPHeadroom,leftOverAlloc)
+
+       #Step 23: ESC Test harness enables DPA activations
+       #deactivated the DPAdeactiveInfo[k] DPA IDs
+       self._sas_admin.TriggerDpaDeactivation(config['DPAdeactiveInfo'][k])
+
+       #activated the DPAactiveInfo[k] DPA IDs
+       #step 24: wait for 240 sec if DPA is activated in step 23 else 15 sec
+       if config['DPAactiveInfo'][k] :
+          TriggerDpaActivation(config['DPAactiveInfo'][k])
+          sleep (240)
+       else:
+          sleep(15)
+
+       # Step 25,26,27 and 28 : 
+       # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
+       dp.heartbeat_request_for_all_grants_and_update_grants()
+       #step 29
+       # TODO: Aggregate interference for DPA
+
+
+       #Step 30:Invoke Aggregate Interference Model
+	
+       cbsd_agg_list = dp.get_all_cbsd_objects_with_atleast_one_grant()
+       
+       for pe in protectedEntities :
+ 
+          #invoking Aggregate Interference API,
+          agg_interface_list = AggregateInterferenceModel(cbsd_agg_list,pe)
+
+          #check MCP.1 DPA
+          #TODO: need to invoke the DPA check
+          #check MCP.1 IAP
+          checkMCP_IAP(pe,agg_interface_list,agg_interference_iap_list,pre_iap_headroom,left_over_alloc)
+   
+  def checkMCP_IAP(self,pe,agg_interface_list,iap_refmodel_margin_list,pre_iap_headroom,left_over_alloc):
+    """ Performs check step """
+    # To Do
+    if ((pe['type'] == 'FSS') or (pe['type'] == 'ESC')): # only one lelement with fifferent frequency
```

Comment:
Please fix typos.

Reply:
Updated,   modified the logic of MCP check